### DISCUSSION TITLE
Publish monorepo with yarn publish

### DISCUSSION THREAD

**Discussion by ayaanqui** [vercel/turborepo#670] [Help] [answered]
Hi, I'm still new to turborepo and I was trying to publish my mono repo with `yarn publish` but I keep getting the following error
```
yarn publish v1.22.15
error Package marked as private, not publishing.
info Visit https://yarnpkg.com/en/docs/cli/publish for documentation about this command.
```

So clearly this happens because yarn workspaces work only on private repo but I wanted to know if there was a way for me to publish my monorepo on npm so I can use some of the existing code in other projects?

Thanks!

**Comment by KrishnaPG** [comment]
Encountered the same problem and looking for help.

**Reply by EarlGeorge** [reply, marked as answer]
first of all, change private to false in your package.json of the package that you are trying to publish or pay some fee for using it as private one, also you will need **github.com/changesets** as well read **npm docs** how to publish a package to the **npm registry** ...

**Comment by ashuvssut** [comment]
For me, in package.json, changing `packageManager` property to `yarn@4.6.0` fixed it (Previously it was `yarn@1.22.22`)


```js
{
	"name": "@scope/my-package",
	"private": false,
	"publishConfig": {
    	"access": "public"
	},
	"packageManager": "yarn@4.6.0"
}
```

_**REASON**_: `yarn@1.22.22` does not support `publishConfig` property of package.json. `yarn@4.6.0` supports `publishConfig` property

------------

If you cannot upgrade your yarn version to `yarn@4.6.0`, you would have to use `npm` instead:-
- Suppose you have a `publish` script in you package.json to publish your package to npm registry
- Here, you should not run `yarn publish`, instead use `npm run publish` because `npm` supports `publishConfig` property.
